tests/FunctionalSshSocksConnectorTest.php

```diff
@@ -29,7 +29,7 @@ public function setUpConnector()
     public function tearDownSSHClientProcess()
     {
         // run loop in order to shut down SSH client process again
-        \React\Async\await(\React\Promise\Timer\sleep(0.001));
+        \React\Async\await(\React\Async\delay(0.001));
```

Comment:
Untested, invalid code. `delay()` returns a `void` and can't be passed to `await()`. Also, `delay()` requires correct Async package version.

Reply:
@clue thanks, you are right with that. I think the whole `async` topic should be discussed in a follow up pull request.

Reply:
@clue the problem was that PHP 5.3 wasn't running green because the `(PromiseTimer/sleep)` function is unknown. PromiseTimer was originally included as a dependency in other ReactPHP components. However, we made the decision to remove this dependency in projects such as [reactphp/socket#305](https://github.com/reactphp/socket/pull/305). The only component which is still using the PromiseTimer component is [ReactPHP MySQL,](https://github.com/friends-of-reactphp/mysq) which doesn't get installed by PHP 5.3 here.

I suggest to include the PromiseTimer dependency in this project as our test suite is currently depending on functions from this component. We don't know when/if PromiseTimer gets removed from the MySQL component, but once this happens the test suite for the other PHP versions will also fail. With my suggested changes we would avoid this happening.